js/tests/unit/modal.js

```diff
@@ -341,24 +349,27 @@ $(function () {
   })
 
   QUnit.test('should adjust the inline body padding when opening and restore when closing', function (assert) {
-    assert.expect(2)
-    var done = assert.async()
-    var $body = $(document.body)
-    var originalPadding = $body.css('padding-right')
-
-    $('<div id="modal-test"/>')
-      .on('hidden.bs.modal', function () {
-        var currentPadding = $body.css('padding-right')
-        assert.strictEqual(currentPadding, originalPadding, 'body padding should be reset after closing')
-        $body.removeAttr('style')
-        done()
-      })
-      .on('shown.bs.modal', function () {
-        var currentPadding = $body.css('padding-right')
-        assert.notStrictEqual(currentPadding, originalPadding, 'body padding should be adjusted while opening')
-        $(this).bootstrapModal('hide')
-      })
-      .bootstrapModal('show')
+    // This test only works when the vertical scrollbar is visible and part of the document (which is not the case for macOS and mobile devices)
+    if ($(this).getScrollbarWidth()) {
```

Comment:
It will fail because no test will be run here you should add an ``else`` condition